src/libraries/System.Security.Cryptography.Csp/src/System/Security/Cryptography/DSACryptoServiceProvider.Unix.cs

```diff
@@ -49,7 +49,16 @@ public DSACryptoServiceProvider(CspParameters parameters)
         }
 
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Security", "CA5351", Justification = "This is the implementation of DSACryptoServiceProvider")]
-        public override byte[] CreateSignature(byte[] rgbHash) => _impl.CreateSignature(rgbHash);
+        public override byte[] CreateSignature(byte[] rgbHash)
+        {
+            if (rgbHash == null)
+                throw new ArgumentNullException(nameof(rgbHash));
+
+            if (rgbHash.Length != SHA1_HASHSIZE)
```

Comment:
This is also a breaking change; but it makes DSACryptoServiceProvider consistent across all platforms.